projects/ngx-datatable/src/lib/types/internal.types.ts

```diff
@@ -0,0 +1,42 @@
+import { TableColumn } from './table-column.type';
+
+export type PinDirection = 'left' | 'center' | 'right';
+
+export interface PinnedColumns {
+  type: PinDirection;
+  columns: TableColumn[];
+}
+
+export interface ColumnGroupWidth {
+  left: number;
+  center: number;
+  right: number;
+  total: number;
+}
+
+export interface OrderableReorderEvent {
+  prevIndex: number;
+  newIndex: number;
+  model: TableColumn;
+}
+
+export interface TargetChangedEvent {
+  newIndex?: number;
+  prevIndex: number;
+  initialIndex: number;
+}
+
+export interface BodyPageEvent {
```

Comment:
This needs to be public since we have public onBodyPage method

Reply:
I actually removed the interface as it is only a number of the page offset. Replaced by number.